amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/common/Parameters.scala

```diff
@@ -51,6 +59,13 @@ case class Parameters(query: Seq[Parameter] = Nil,
 
   private def addPayloads(global: Seq[Payload], inner: Seq[Payload]): Seq[Payload] = global ++ inner
 
+  private def findDuplicateParametersBetween(global: Seq[Parameter], inner: Seq[Parameter]): List[Parameter] = {
```

Comment:
A few thoughts on this:
- It can be coded more expressively:
```scala
global.foldLeft { (list, parameter) => 
      inner.find(_.name.value() == parameter.name.value()).getOrElse(Nil) :: list
    }```
- You are doing a O(n^2) search for names. Maybe it would be useful to index parameters by name like a Map to make ir more efficient. It is true that there won't be a lot of parameters in an API but nonetheless the performance improvement is useful and it is trivial to implement.